<!DOCTYPE html>
<html><head><title>FlowControlException</title><link rel="stylesheet" href="../../styles.css"/><script src="../../scripts.js"></script></head><body onload="ro();">
<div class="rH">6 types derived from FlowControlException</div><div class="rA">System.Management.Automation (6)</div><div class="rG" id="System.Management.Automation"><div class="rF"><div class="rN">engine\lang\parserutils.cs (6)</div>
<a href="../engine/lang/parserutils.cs.html#41"><b>41</b>public abstract class LoopFlowException : <i>FlowControlException</i></a>
<a href="../engine/lang/parserutils.cs.html#131"><b>131</b>internal class ReturnException : <i>FlowControlException</i></a>
<a href="../engine/lang/parserutils.cs.html#144"><b>144</b>public class ExitException : <i>FlowControlException</i></a>
<a href="../engine/lang/parserutils.cs.html#170"><b>170</b>internal class ExitNestedPromptException : <i>FlowControlException</i></a>
<a href="../engine/lang/parserutils.cs.html#177"><b>177</b>public sealed class TerminateException : <i>FlowControlException</i></a>
<a href="../engine/lang/parserutils.cs.html#186"><b>186</b>internal class StopUpstreamCommandsException : <i>FlowControlException</i></a>
</div>
</div>
<div class="rH">31 references to FlowControlException</div><div class="rA">System.Management.Automation (31)</div><div class="rG" id="System.Management.Automation"><div class="rF"><div class="rN">engine\CommandProcessorBase.cs (1)</div>
<a href="../engine/CommandProcessorBase.cs.html#801"><b>801</b>|| e is <i>FlowControlException</i></a>
</div>
<div class="rF"><div class="rN">engine\CoreAdapter.cs (3)</div>
<a href="../engine/CoreAdapter.cs.html#612"><b>612</b>catch (<i>FlowControlException</i>) { throw; }</a>
<a href="../engine/CoreAdapter.cs.html#4155"><b>4155</b>if (ex.InnerException is <i>FlowControlException</i> || ex.InnerException is ScriptCallDepthException)</a>
<a href="../engine/CoreAdapter.cs.html#4175"><b>4175</b>catch (<i>FlowControlException</i>) { throw; }</a>
</div>
<div class="rF"><div class="rN">engine\hostifaces\LocalPipeline.cs (5)</div>
<a href="../engine/hostifaces/LocalPipeline.cs.html#276"><b>276</b>private <i>FlowControlException</i> InvokeHelper()</a>
<a href="../engine/hostifaces/LocalPipeline.cs.html#278"><b>278</b><i>FlowControlException</i> flowControlException = null;</a>
<a href="../engine/hostifaces/LocalPipeline.cs.html#451"><b>451</b>catch (<i>FlowControlException</i> e)</a>
<a href="../engine/hostifaces/LocalPipeline.cs.html#514"><b>514</b>catch (<i>FlowControlException</i>)</a>
<a href="../engine/hostifaces/LocalPipeline.cs.html#591"><b>591</b><i>FlowControlException</i> flowControlException = InvokeHelper();</a>
</div>
<div class="rF"><div class="rN">engine\lang\parserutils.cs (5)</div>
<a href="../engine/lang/parserutils.cs.html#1282"><b>1282</b>catch (<i>FlowControlException</i>)</a>
<a href="../engine/lang/parserutils.cs.html#1437"><b>1437</b>catch (<i>FlowControlException</i>)</a>
<a href="../engine/lang/parserutils.cs.html#1472"><b>1472</b>catch (<i>FlowControlException</i>)</a>
<a href="../engine/lang/parserutils.cs.html#1520"><b>1520</b>/// &lt;exception cref=&quot;<i>FlowControlException</i>&quot;&gt;Internal exception from a flow control statement.&lt;/exception&gt;</a>
<a href="../engine/lang/parserutils.cs.html#1633"><b>1633</b>catch (<i>FlowControlException</i>)</a>
</div>
<div class="rF"><div class="rN">engine\lang\scriptblock.cs (4)</div>
<a href="../engine/lang/scriptblock.cs.html#396"><b>396</b>/// &lt;exception cref=&quot;<i>FlowControlException</i>&quot;&gt;An internal (non-public) exception from a flow control statement.&lt;/exception&gt;</a>
<a href="../engine/lang/scriptblock.cs.html#517"><b>517</b>/// &lt;exception cref=&quot;<i>FlowControlException</i>&quot;&gt;An internal (non-public) exception from a flow control statement.&lt;/exception&gt;</a>
<a href="../engine/lang/scriptblock.cs.html#859"><b>859</b>/// &lt;exception cref=&quot;<i>FlowControlException</i>&quot;&gt;Internal exception from a flow control statement.&lt;/exception&gt;</a>
<a href="../engine/lang/scriptblock.cs.html#917"><b>917</b>/// &lt;exception cref=&quot;<i>FlowControlException</i>&quot;&gt;Internal exception from a flow control statement.&lt;/exception&gt;</a>
</div>
<div class="rF"><div class="rN">engine\MshCmdlet.cs (3)</div>
<a href="../engine/MshCmdlet.cs.html#684"><b>684</b>/// &lt;exception cref=&quot;<i>FlowControlException</i>&quot;&gt;&lt;/exception&gt;</a>
<a href="../engine/MshCmdlet.cs.html#699"><b>699</b>/// &lt;exception cref=&quot;<i>FlowControlException</i>&quot;&gt;&lt;/exception&gt;</a>
<a href="../engine/MshCmdlet.cs.html#797"><b>797</b>/// &lt;exception cref=&quot;<i>FlowControlException</i>&quot;&gt;&lt;/exception&gt;</a>
</div>
<div class="rF"><div class="rN">engine\MshMemberInfo.cs (3)</div>
<a href="../engine/MshMemberInfo.cs.html#1844"><b>1844</b>catch (<i>FlowControlException</i> e)</a>
<a href="../engine/MshMemberInfo.cs.html#1875"><b>1875</b>catch (<i>FlowControlException</i> e)</a>
<a href="../engine/MshMemberInfo.cs.html#2403"><b>2403</b>catch (<i>FlowControlException</i> e)</a>
</div>
<div class="rF"><div class="rN">engine\parser\Compiler.cs (3)</div>
<a href="../engine/parser/Compiler.cs.html#848"><b>848</b>s_catchFlowControl = Expression.Catch(typeof(<i>FlowControlException</i>), Expression.Rethrow());</a>
<a href="../engine/parser/Compiler.cs.html#2168"><b>2168</b>catch (<i>FlowControlException</i>)</a>
<a href="../engine/parser/Compiler.cs.html#4488"><b>4488</b>Expression.Catch(typeof(<i>FlowControlException</i>), Expression.Rethrow(tryBlock.Type)),</a>
</div>
<div class="rF"><div class="rN">engine\runtime\Operations\MiscOps.cs (3)</div>
<a href="../engine/runtime/Operations/MiscOps.cs.html#2054"><b>2054</b>if ((exception is <i>FlowControlException</i> ||</a>
<a href="../engine/runtime/Operations/MiscOps.cs.html#3367"><b>3367</b>catch (<i>FlowControlException</i>)</a>
<a href="../engine/runtime/Operations/MiscOps.cs.html#3401"><b>3401</b>catch (<i>FlowControlException</i>)</a>
</div>
<div class="rF"><div class="rN">engine\ScriptCommandProcessor.cs (1)</div>
<a href="../engine/ScriptCommandProcessor.cs.html#581"><b>581</b>catch (<i>FlowControlException</i>)</a>
</div>
</div>
</body></html>